src/rest/routes/deferred/deferred-installation-url.ts

```diff
@@ -0,0 +1,40 @@
+import { Request, Response } from "express";
+import { errorWrapper } from "../../helper";
+import { GetDeferredInstallationUrl } from "rest-interfaces";
+import { BaseLocals } from "..";
+import {
+	registerSubscriptionDeferredInstallPayloadRequest,
+	SubscriptionDeferredInstallPayload
+} from "services/subscription-deferred-install-service";
+import { envVars } from "config/env";
+import { InvalidArgumentError } from "config/errors";
+
+export const DeferredInstallationUrlRoute = errorWrapper("GetDeferredInstallationUrl", async function DeferredInstallationUrlRoute(req: Request, res: Response<GetDeferredInstallationUrl, BaseLocals>) {
+	const { gitHubInstallationId, gitHubOrgName } = req.query;
+	const { installation } = res.locals;
+
+	if (!gitHubInstallationId) {
+		req.log.warn("Missing gitHubInstallationId in query");
+		throw new InvalidArgumentError("Missing gitHubInstallationId in query");
+	}
+
+	if (!gitHubOrgName) {
+		req.log.warn("Missing gitHubOrgName in query");
+		throw new InvalidArgumentError("Missing gitHubOrgName in query");
+	}
+
```

Comment:
is it possible for res.locals.installation to not be available at this point?

Reply:
Nop, `res.locals.installation` should always be here because this route passes through `JwtHandler` middleware, which parses the JWT token and saves the `installation` to `res.locals`.